docs/sql-reference/10-sql-commands/00-ddl/02-user/04-user-set-role.md

```diff
@@ -0,0 +1,38 @@
+---
+title: SET ROLE
+sidebar_position: 5
+---
+
+Switches the active role for a session, and the currently active role can be viewed using the [SHOW ROLES](04-user-show-roles.md) command, with the `is_current` field indicating the active role.
+
+## Syntax
+
+```sql
+SET ROLE <role_name>
+```
+
+## Examples
+
+```sql
+SHOW ROLES;
+
+┌───────────────────────────────────────────────────────┐
+│    name   │ inherited_roles │ is_current │ is_default │
+├───────────┼─────────────────┼────────────┼────────────┤
+│ developer │               0 │ false      │ false      │
+│ public    │               0 │ false      │ false      │
+│ writer    │               0 │ true       │ true       │
+└───────────────────────────────────────────────────────┘
+
+SET ROLE developer;
+
+SHOW ROLES;
+
+┌───────────────────────────────────────────────────────┐
+│    name   │ inherited_roles │ is_current │ is_default │
+├───────────┼─────────────────┼────────────┼────────────┤
+│ developer │               0 │ true       │ false      │
```

Comment:
I manually updated this row to 'true' for the expected result, as the command might have a bug, as we discussed.